src/incubator/WheelPicker/index.tsx

```diff
@@ -2,90 +2,164 @@
 // TODO: Support style customization
 // TODO: Support control of visible items
 import _ from 'lodash';
-import React, {useCallback, useRef} from 'react';
-import {TextStyle, FlatList, StyleProp} from 'react-native';
+import React, {useCallback, useRef, useState, useMemo} from 'react';
+import {TextStyle, FlatList, StyleProp, NativeSyntheticEvent, NativeScrollEvent} from 'react-native';
 import Animated from 'react-native-reanimated';
-import {onScrollEvent, useValues} from 'react-native-redash';
-
+import {useValues} from 'react-native-redash';
+import {Colors} from '../../../src/style';
 import View from '../../components/view';
+import Fader, {FaderPosition} from '../../components/fader';
 import {Constants} from '../../helpers';
 
 import Item, {ItemProps} from './Item';
 
 const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
 
 export interface WheelPickerProps {
+  /**
+   * Data source for WheelPicker
+   */
   items?: ItemProps[];
+  /**
+   * Describe the height of each item in the WheelPicker
+   */
   itemHeight?: number;
-  itemTextStyle?: StyleProp<TextStyle>;
-  onChange: (item: ItemProps, index: number) => void;
+  /**
+   * TextStyle for the focused row
+   */
+  activeItemTextStyle?: StyleProp<TextStyle>;
+  /**
+   * TextStyle for other, non-focused rows
+   */
+  inactiveItemTextStyle?: StyleProp<TextStyle>;
+  /**
+   * Event, on active row change
+   */
+  onChange: (index: number, item?: ItemProps) => void;
 }
 
-const WheelPicker = ({items, itemHeight = 48, itemTextStyle}: WheelPickerProps) => {
-  const height = itemHeight * 4;
-  const scrollview = useRef<Animated.ScrollView>();
+type WrappedItem = {
+  isActive: boolean;
+} & ItemProps;
+
+const WheelPicker = ({items, itemHeight = 48, activeItemTextStyle, inactiveItemTextStyle, onChange: onChangeEvent}: WheelPickerProps) => {
+  const height = itemHeight * 5;
+  const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
-  const onScroll = onScrollEvent({y: offset});
+  const activeIndex = useRef<number>(0);
+
+  const wrapItems = (items?: ItemProps[]): WrappedItem[] => {
+    return _.map(items, (item, index: number) => {
+      return {...item, isActive: activeIndex.current === index};
+    });
+  };
```

Comment:
what's the purpose of `wrapItems`? 
I see you only use it for the keyExtractor, is that really necessary? 
What does it solve?